coil-base/src/main/java/coil/network/NetworkObserver.kt

```diff
@@ -1,51 +1,146 @@
 package coil.network
 
+import android.Manifest.permission.ACCESS_NETWORK_STATE
+import android.annotation.SuppressLint
+import android.content.BroadcastReceiver
 import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import android.os.Build.VERSION.SDK_INT
 import android.util.Log
+import androidx.annotation.MainThread
+import androidx.annotation.RequiresApi
+import androidx.core.content.getSystemService
+import coil.network.NetworkObserver.Listener
 import coil.util.Logger
+import coil.util.isPermissionGranted
 import coil.util.log
 
 /**
- * This class caches the network state from [NetworkObserverStrategy].
+ * Observes the device's network state and calls [Listener] if any state changes occur.
  *
- * Instances should be cleaned up with [shutdown].
+ * This class provides a raw stream of updates from the network APIs. The [Listener] can be
+ * called multiple times for the same network state.
  */
-internal class NetworkObserver(
```

Comment:
This diff isn't accurate as `NetworkObserver` was deleted and `NetworkObserverStrategy` was renamed to `NetworkObserver`.